File: 800/Hit the Lottery/htl.cpp
// https://codeforces.com/problemset/problem/996/A

#include <iostream>
#include <climits>
using namespace std;

int Minimum(int arr[6]) {
    int min = arr[0];
    for(int i = 1; i < 6; i++) {
        if(arr[i] < min) {
            min = arr[i];
        }
    }
    return min;
}

// Main driver
int main() {
    int n;
    cout << "Enter the number: ";
    cin >> n;
    cout << "The number is: " << n << endl;
    int arr[6] = {1, 5, 10, 20, 50, 100};
    int diffarr[6], count = 0;
    while(n != 1 || n != 5 || n != 10 || n != 20 || n != 50 || n != 100 ) {
        if(n == 1 || n == 5 || n == 10 || n == 20 || n == 50 || n == 100) {
            count = count + 1;
        }
        else {
            for(int i = 0; i < 6; i++) {
                diffarr[i] = n - arr[i];
                if(diffarr[i] < 0) {
                    diffarr[i] = INT_MAX;
                }
            }
            cout << "The difference array is: ";
            for(int i = 0; i < 6; i++) {
                cout << diffarr[i] << " ";
            }
            cout << "\n";
            int temp = n - Minimum(arr);
            int flag = n / temp;
            count = count + flag;
            n = n - flag * (n - Minimum(arr));
        }
    }
    cout << "The minimum of denominations are: " << count << endl;
    return 0;
}

File: 800/Hit the Lottery/htl.py
# https://codeforces.com/problemset/problem/996/A

def minimum(a = []):
  min = a[0]
  for i in range(1, 6, 1):
    if(a[i] < min):
      min = a[i]
  return min

n = int(input("Enter the number: "))
print("The number is: ", n)
d = [1, 5, 10, 20, 50, 100]
print("The denomination array is: ", d)
diffarr = [];
count = 0
while n != 1 or n != 5 or n != 10 or n!= 20 or n != 50 or n != 100:
  for i in range(len(d)):
    diffarr[i] = n - d[i]
    if diffarr[i] < 0:
      diffarr[i] = 1000000000000
  
  print("The difference array is: ", diffarr)
  #print("\n")
  temp = n - minimum(d);
  flag = n / temp;
  count = count + flag;
  n = n - flag * (n - minimum(d));
  
if n == 1 or n == 5 or n == 10 or n == 20 or n == 50 or n == 100:
  count = count + 1

print("The minimum number of denominations are: ", count)

